### PR TITLE
OCPBUGS-30658: pkg/cli/admin/upgrade: --to-latest help "next" -> "latest"

### PR DESCRIPTION
**Pull request by wking**
Making it more clear which release would be selected.

### CODE REVIEW

File: pkg/cli/admin/upgrade/upgrade.go [hunk 1]
```diff
@@ -104,7 +104,7 @@ func New(f kcmdutil.Factory, streams genericiooptions.IOStreams) *cobra.Command
 	flags := cmd.Flags()
 	flags.StringVar(&o.To, "to", o.To, "Specify the version to upgrade to. The version must be on the list of available updates.")
 	flags.StringVar(&o.ToImage, "to-image", o.ToImage, "Provide a release image to upgrade to.")
-	flags.BoolVar(&o.ToLatestAvailable, "to-latest", o.ToLatestAvailable, "Use the next available version.")
+	flags.BoolVar(&o.ToLatestAvailable, "to-latest", o.ToLatestAvailable, "Use the latest (largest Semantic Version) available version.")
```
Comment (petr-muller) [inline pkg/cli/admin/upgrade/upgrade.go:107]: "highest" would sound more precise to me but I'm not the native english speaker here so I trust your judgement, just unhold :D

/lgtm
/hold

**Review by petr-muller — APPROVED** [review]

File: pkg/cli/admin/upgrade/upgrade.go (hunk 1 above)
Comment (wking) [reply pkg/cli/admin/upgrade/upgrade.go]: I pushed "largest" -> "highest" in c04481d421fc -> 56db34e82d1c.

/hold cancel

Comment (JianLi-RH) [reply pkg/cli/admin/upgrade/upgrade.go]: Here is the new output compared to the old one:
```
[jianl@jianl-thinkpadt14gen4 417]$ diff -uNb -B2 <(oc adm upgrade --help) <(./oc adm upgrade --help) | tail -n +4
 
     --to-latest=false:
-	Use the next available version.
+	Use the latest (highest Semantic Version) available version.
 
     --to-multi-arch=false:
[jianl@jianl-thinkpadt14gen4 417]$
```

I am ok with the change, a minor suggestion, maybe we just need to say `highest Semantic` in the parentheses.

**Comment by JianLi-RH** [comment]
so far LGTM, cc @jiajliu for a final review. 
```
[jianl@jianl-thinkpadt14gen4 417]$ ./oc version
Client Version: 4.18-dev
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: 4.17.0-0.nightly-2024-08-29-171003
Kubernetes Version: v1.30.3
[jianl@jianl-thinkpadt14gen4 417]$ diff -uNb -B2 <(oc adm upgrade --help) <(./oc adm upgrade --help) | tail -n +4
 
     --to-latest=false:
-	Use the next available version.
+	Use the latest (highest Semantic Version) available version.
 
     --to-multi-arch=false:
[jianl@jianl-thinkpadt14gen4 417]$
```

File: pkg/cli/admin/upgrade/upgrade.go (hunk 1 above)
Comment (JianLi-RH) [reply pkg/cli/admin/upgrade/upgrade.go]: btw do we really need to capitalize the first letter of `Semantic Version`?

**Review by petr-muller — APPROVED** [review]
/lgtm
/hold

Re-adding lgtm, and hold to allow QE finish convos here.

**Comment by wking** [comment]
[Held for QE][1] and QE is [on][2] [board][3].

/hold cancel

[1]: https://github.com/openshift/oc/pull/1861#pullrequestreview-2271815967
[2]: https://github.com/openshift/oc/pull/1861#issuecomment-2320260381
[3]: https://github.com/openshift/oc/pull/1861#issuecomment-2320728644

**Pull request merged**